akka-typed/src/main/scala/akka/typed/cluster/ClusterSingletonManager.scala

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster
+
+import akka.actor.NoSerializationVerificationNeeded
+import akka.cluster.ClusterSettings.DataCenter
+import akka.cluster.singleton.{ ClusterSingletonManagerSettings, ClusterSingletonProxy, ClusterSingletonProxySettings, ClusterSingletonManager ⇒ OldSingletonManager }
+import akka.typed.internal.adapter.{ ActorRefAdapter, ActorSystemAdapter }
+import akka.typed.scaladsl.adapter._
+import akka.typed.{ ActorRef, ActorSystem, Behavior, Extension, ExtensionId, Props }
+import com.typesafe.config.Config
+
+import scala.concurrent.duration.FiniteDuration
+
+object ClusterSingletonSettings {
+  def apply[A](
+    system: ActorSystem[_]
+  ): ClusterSingletonSettings = ???
+
+  def fromConfig[A](
+    config: Config
+  ): ClusterSingletonSettings = ???
+}
+
+final class ClusterSingletonSettings(
+  val role:                            Option[String],
+  val dataCenter:                      Option[DataCenter],
+  val singletonIdentificationInterval: FiniteDuration,
+  val removalMargin:                   FiniteDuration,
+  val handOverRetryInterval:           FiniteDuration,
+  val bufferSize:                      Int) extends NoSerializationVerificationNeeded {
+
+  private[akka] def toManagerSettings: ClusterSingletonManagerSettings = ???
+  private[akka] def toProxySettings: ClusterSingletonProxySettings = ???
+
+}
+
+object ClusterSingletonManager extends ExtensionId[ClusterSingletonManager] {
+
+  override def createExtension(system: ActorSystem[_]): ClusterSingletonManager =
+    new ClusterSingletonManagerImpl(system)
+}
+
+private[akka] class ClusterSingletonManagerImpl(system: ActorSystem[_]) extends ClusterSingletonManager {
+  require(system.isInstanceOf[ActorSystemAdapter[_]], "only adapted actor systems can be used for the typed cluster singleton")
+
+  private val untypedSystem = ActorSystemAdapter.toUntyped(system)
+
+  override def spawn[A](
+    behavior:           Behavior[A],
+    name:               String,
+    props:              Props,
+    settings:           ClusterSingletonSettings,
+    terminationMessage: Option[A]                = None) = {
```

Comment:
the terminationMessage should probably not be optional, when there is no poison pill